transaction/issuance.go

```diff
@@ -0,0 +1,126 @@
+package transaction
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/vulpemventures/fastsha256"
+	"github.com/vulpemventures/go-elements/confidential"
+	"github.com/vulpemventures/go-elements/internal/bufferutil"
+)
+
+// IssuanceEntity defines one of the fields of the issuance contract
+type IssuanceEntity struct {
+	Domain string `json:"domain"`
+}
+
+// IssuanceContract defines the structure of the Ricardian contract of the issuance
+type IssuanceContract struct {
+	Name      string         `json:"name"`
+	Ticker    string         `json:"ticker"`
+	Version   uint           `json:"version"`
+	Precision uint           `json:"precision"`
+	Entity    IssuanceEntity `json:"entity"`
+}
+
+// TxIssuance defines the type for Issuance field in TxInput
+type TxIssuance struct {
+	AssetBlindingNonce []byte
+	AssetEntropy       []byte
+	AssetAmount        []byte
+	TokenAmount        []byte
+}
+
+// NewTxIssuance returns a new empty issuance instance
+func NewTxIssuance(assetAmount, tokenAmount uint64) (*TxIssuance, error) {
```

Comment:
We should pass the precision as parameter here also and then we would make to check in the GenerateEntropy method that the asset amount provided and precision together are correct.

Reply:
do you know if the precision affects also the token amount or just the asset amount?

Reply:
I guess that the reissuance tokens could be maximum 21M*10**8 and not more/less. But thats' a good question

Reply:
![Screen Shot 2020-06-09 at 4 04 05 PM](https://user-images.githubusercontent.com/3596602/84157317-e563ce00-aa6a-11ea-846c-42e340ca3d76.png)

@altafan

Reply:
Consider that the fields of the `TxIssuance` type are just the ones that will be encoded in the transaction. Note that the eventual range proofs for confidential asset and token are not part of this type because they are encoded together with the witnesses and treated separatedly.

I agree that it would be nice to have the precision here, but I don't think we should add a non-encoded field to the TxIssuance type.

Reply:
I'm not saying we should add the field, but we should be able to check somehow tha the suer si providingi the right asset amount and the right precision. Maybe we should do later on a convenience high level function that call these?

Reply:
@altafan